worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java

```diff
@@ -227,6 +234,26 @@ private int processHandles(List<Future<EobSearchResult>> futureHandles, Progress
         return numberOfEobs;
     }
 
+    void addMbiIdToEobs(List<ExplanationOfBenefit> eobs, Map<String, PatientDTO> patients) {
+        if (eobs == null || eobs.isEmpty()) {
+            return;
+        }
+        // Get first EOB Bene ID
+        ExplanationOfBenefit eob = eobs.get(0);
+        String mbi = null;
+        String benId = getPatientIdFromEOB(eob);
+        if (benId != null) {
+            mbi = patients.get(benId).getMbiId();
+        }
+        Extension extension = createExtension(mbi);
+        eobs.stream().forEach(e -> e.addExtension(extension));
+    }
+
+    Extension createExtension(String mbi) {
+        Identifier identifier = new Identifier().setSystem(MBI_ID).setValue(mbi);
+        return new Extension().setUrl(ID_EXT).setValue(identifier);
```

Comment:
Do we want to move MBI_ID and the other ids to somewhere more central like a constants class?

Reply:
Probably and I was thinking there may even be values for it in the hapi library but I'm not losing any sleep over this.